### DISCUSSION TITLE
Information about SDRPlay settings

### DISCUSSION THREAD

**Discussion by jaymot123** [szpajder/dumphfdl#25] [General]
I sent a support request to SDRPlay asking some technical questions about sample rates and gain values and this is the response, which may be helpful to others:

**Question:** What unit of measurement do the sample rates use? MHz or what?

**Answer:** The units for sample rate are MHz. This is the raw sample rate for the ADCs. The sample rate can only sit in the range of 2 - 10 MHz. To obtain sample rates of less than 2 MHz, decimation has to be used as a post processing step.

**Question:** What exactly is the AGC setpoint and what does it do?

**Answer:** The AGC set point refers specifically to the IF AGC and is the target level that the total RMS signal sits below the full scale range of the ADCs. So for example, a set point of -30 would aim to set the total composite RMS signal at a level that was 30 dB below the full scale range of the ADC. The reason for this is that with wideband SDRs, the peak to average ratio can be very high and values in excess of 20 dB are common. It is important to avoid 'clipping' on the ADCs (clipping is where the peaks of the signal exceed the full scale range) or severe distortion will occur. As a result, we recommend a value for the set point of being in the range of -20 to -30 dBFs. If you go lower than -30 dBFs, then you are throwing away ADC dynamic range. If you go much higher, then you risk seeing intermodulation and other overload artefacts.

**Question:** Dumphfdl requires soapysdr, and running the command "SoapySDRUtil --probe=driver=sdrplay" shows (among other information): 
```
    Full gain range: [0, 48] dB
    IFGR gain range: [20, 59] dB
    RFGR gain range: [0, 9] dB
```
What are IFGR and RFGR and how do they differ from the full gain range? Please explain these different types of gain in some detail so I can better understand the internals of this radio.

**Answer:** The RSP1A has two regions of gain control. IF Gain and RF Gain. Gain control is achieved via the application of distributed switched attenuators and the gain for both the RF stages and the IF stages can be adjusted in a number of discrete steps, each of which is assigned a step number within the API. Because the gain adjustment is achieved with attenuators, SoapSDRplay represents this as 'Gain Reduction' or GR. So IFGR is IF Gain reductions and RFGR is RF Gain reduction.

At HF, the RF gain can be adjusted in 10 discrete steps, (step 0 - 9) and the IF Gain adjustment has 59 possible steps, but the first 19 are not available within the API as they would always result in ADC clipping, so steps 20 - 59 are available for gain adjustment.

The total number of steps is therefore 49 (steps 0 - 48), which is what is represented by the full gain. However, because the RFGR and IFGR are separate, the total number of gain step possibilities is actually 390.

The API contains gain calibration tables that allows the conversion of a specific combination of gain control steps to be mapped to a specific gain and any given frequency, but SoapySDRplay does not use these, so these steps should be considered to be illustrative rather than absolute measurements of gain.

In final point to note is that the number of available RF gain reduction steps does vary depending upon the frequency. 10 steps is the maximum number of steps and this occurs at HF and below.

If the IF AGC is enabled, then the control of the IF gain reduction is automatic, leaving the user just to adjust the RF gain reduction to avoid overload. We recommend leaving the IF AGC enabled.

**Question**: I'm using my RSP1A with a K-180WLA active loop HF antenna which has a 20dBi amplifier. (I've never gotten the SDR to work very well with the dipole I made from scrap wire with each leg being slightly longer than 6 meters, connected to a 9:1 balun and RG58 coax, but the loop works very well indeed using the same coax. I think the dipole's picking up too much local noise from nearby electrical appliances. Sadly, I don't have a large enough lot to put the dipole very far from my house.) Can you recommend approximate gain settings so that I don't overload the front end of the radio and increase the noise floor so much that reception of wanted signals is suboptimal? I understand that you can't possibly tell me exactly what my gain should be for my antenna and location but if you could perhaps suggest a range of values that I could try?

**Answer:** Finally regarding the 'optimum' value for gain setting, as to (sic) have correctly surmised, it is impossible to say as it depends upon current reception conditions for each individual user and this can and does vary hugely.

You want to set the gain as high as possible whilst avoiding ADC overload to get the best sensitivity and dynamic range.The key is to use a visual indication of whether the gain is set too high or not. I am not familiar with dumphfdl, but if there is no spectrum representation or waterfall, your best approach might be to use CubicSDR to tune to the frequencies of interest and check the spectrum display and waterfall for possible overload. This should allow you to get an idea of what are the optimum RF gain settings for a give frequency for your specific set up. You would then close CubicSDR and start dumphfdl and apply those gain settings.

If dumphfdl allows you to enable the IF AGC, I would do so. If it allows you to see that the IFGR value has been set to by the IF AGC, then all the better. If you see the IFGR approach 59, then you should consider reducing the RF gain a step (increasing RFGR). If the IF AGC is setting is well below 59, then you should consider increasing the RF gain (reducing RFGR).

**Comment by szpajder** [comment]
Thank you for sharing this.

I also use RSP1A and run dumphfdl on it. Before that I used it for casual wave surfing with GQRX. I don't use AGC and I set the gain manually as follows:

- set RFGR to 0 (ie. max gain)
- set IFGR to get good signal levels with no overload
- if max IFGR (59) is not high enough to get rid of overloads, then increase RFGR by one notch and tweak IFGR again
- rinse and repeat

RFGR=0,IFGR=54 gives me good results most of the time. I only bump up RFGR to 1 when I'm in the mood for listening to MW stations where signal levels are often very high. HFDL runs fine on RFGR=0 on all bands. Of course this depends on the antenna and the local RF environment (interference level or the like). Note that I generally operate on low sampling rates (not higher than 2 Msps). Given that RSP1A supports rates up to 10 Msps, it might be tempting to crank it up to the max to cover as many HFDL bands as possible. However if you want DX, then this is usually a suboptimal choice for two reasons:

- the larger the bandwidth, the more unwanted strong signals in it (ie. broadcast stations). This limits your ability of using high gains (ie. low GR values) because those strong signals will then overload the radio and spill across the whole band. You therefore need to keep GRs high enough to prevent that from happening, however this makes the levels of all other signals (including HFDL) lower as well, which might cause them to blend with the background noise and become undecodable.

- the larger the sampling rate, the lower the bit resolution of the ADC. From the specs:

```
14bit  2MSPS – 6.048MSPS
12bit  6.048MSPS – 8.064MSPS
10bit  8.064MSPS – 9.216MSPS
8bit   >9.216MSPS
```

This means that at 10 Msps the ADC operates at the same resolution as an RTL dongle. This limits your dynamic range (ie. ability to receive very strong and very weak signals simultaneously) and increases quantization noise, which in turn hinders decoding of weakest HFDL messages.

Monitoring many HFDL bands using a single receiver with a high sampling rate is also a waste of electricity, since there is a lot of the data to process and most of this data is useless afterwards. A better choice is to use several receivers with as low sampling rate as possible. So what I currently use is one RSP1A at 2 Msps which covers up to 2 HFDL bands (5+6 MHz or 8+10 MHz or 10+11 MHz) and two Airspy HF+ at 192 or 384 ksps, each one covering a single band. This allows monitoring of four bands on an Odroid XU4 and a Raspberry Pi ver 3. From experience I can also say that RPi v4 can easily handle four dumphfdl instances on four Airspy HF+ receivers with a load average below 1.0. The only problem to address was the current limit on USB ports - three radios were OK, but the fourth one caused overcurrents and port resets, so it was necessary to move two radios to an external powered hub.

**Reply by jaymot123** [reply]
So that's why you recommended a sample rate of 6MHz earlier. My wide spreads of frequencies with a 10MHz sample rate may also be why I rarely see any aircraft in VRS beyond Asia, although I did get some in the Persian Gulf area this morning and saw one in Brazil the other day. I only see far-away ones early in the morning at around sunrise.  Other people in various discussion boards are seeing airplanes in VRS in Asia, North and South America, Europe and over the oceans, at least on good days for HP propagation. (Of course they aren't necessarily using RSP1As and active small-loop antennas either.)

The fact that I've been able to see any aircraft at all shows what a great SDR the RSP1A is though, even when running as only 8-bit. I tested dumphfdl yesterday with my new RTL-SDR Blog v3 dongle, and as I expected it couldn't receive anything at all.

I'll try again with a lower spread of frequencies and 6MHz sampling (or less) so I can take advantage of the SDR's higher bit rates. I also think I'll extend VRS's JAERO time-outs so it shows planes with messages that are up to around 12 hours old so I can check it in the morning and see what it's received during the night, so I don't have to stay up all night long to find out. I'll also try your recommended IFGR and RFGF settings.

**Reply by szpajder** [reply]
> I only see far-away ones early in the morning at around sunrise. Other people in various discussion boards are seeing airplanes in VRS in Asia, North and South America, Europe and over the oceans

This shouldn't be a problem these days. Solar flux is above 100 most of the time, which means good ionisation. If you only see local traffic, then maybe it's a QRM issue? If most of HFDL signals are buried in man-made noise, then boosting AGC bit resolution probably won't help much. I suggest doing a thorough inspection of the spectrum in various HFDL bands on various times of day (and night). Background should be clean and mostly uniform (but its level goes up naturally, as you go down with frequency - this is normal). If there is interference, then it's better if it's strong rather than weak. Why? Because if it's strong, then the source is probably in your own house, so you can track it down and eliminate it.

Defective installation might also be the culprit. Poor connections are a path for noise ingress. Think of things like: unstable / oxidized / rusty connections between the loop and the head amplifier, connectors that are either poorly crimped or not tightened properly, very old or very cheap coax with poor screening, etc.

> I also think I'll extend VRS's JAERO time-outs so it shows planes with messages that are up to around 12 hours old so I can check it in the morning and see what it's received during the night, so I don't have to stay up all night long to find out.

Note that you can also record all position messages sent by dumphfdl to VRS with `--output decoded:basestation:file:path=/some/file`. You will then have an archive of all valid position reports received and you'll be able to do any kind of analyses on this data. I often just sort the file by latitude or longitude to quickly check for most extreme DXes received opn a given day or week (`sort -nt, -k15 file.log` for latitude, `-k16` for longitude). Crude, but it works.

**Reply by jaymot123** [reply]
My noise floor seems to be not too bad with the gain elements that I'm using. The S/N from aircraft messages is typically 3dB or so. I did pick up a couple of planes in the eastern Pacific near Hawaii and one in Brazil yesterday afternoon as the sun was starting to go down and the greyline terminator approached my location (Philippines), but after that the farthest planes I saw were over India. I've just figured out how to power my K-180WLA active loop antenna's amplifier from the RSP1A's bias-T so it will get 4.7VDC instead of 3.7VDC from its PS box's internal 18650 battery and I'll test it again today and tonight and see if that gives me any improvement.

Edit: scratch that idea, the antenna's bias-T box contains a DC boost circuit that I didn't know about earlier, which raises the 3,7VDC to  around 11 volts (and is also somewhat noisy. They should have used 3 18650s in series and put the DC boost on the charging circuit.)

**Reply by szpajder** [reply]
K-180WLA head amp is very wideband (up to 180 MHz) and so it is often susceptible to overload by strong FM stations. Make sure you don't have this problem.

Illustration: https://www.youtube.com/watch?v=R4-KOLQzNYU

I had this issue with a LZ1AQ head amp. Solved with a DIY FM notch filter plugged into the loop terminals (ie. at the amp input).

Results (left is before, right is after):

![image](https://user-images.githubusercontent.com/4560159/154868645-aa963651-2e75-4348-abe0-df1987e6b221.png)

If such a noise hump happens to appear in the middle of a HFDL band, then one may forget about any kind of DX.

**Comment by jaymot123** [comment]
Upon watching my multitailed log some more I believe I was wrong earlier and that my issue is indeed an often poor SNR, probably due to my noisy location. The power lines here are in very poor condition and the recent typhoon made them even worse. When the electricity was off for weeks following that storm I had very good shortwave reception on my Tecsun PL-310ET portable radio, but ever since power was restored I can't hear very many stations on it at all, mostly just noise, so that's another clue. The K-180WLA antenna does no noise-cancelling nor does my homemade dipole (which is also noisy and gives me low signal strength.) I don't seem to have any humps or spikes in the FFT in SDR++ due to FM broadcast stations, just a rather high noise floor compared to your images (for which, thank you.)

I've ordered an Airspy YouLoop passive loop antenna which per a [review](https://qrznow.com/youloop-hf-loop-antenna/) on qrznow.com is supposed to have good electrical noise-cancelling qualities in the HF bands due both to its Mobius loop-like design and its isolation transformer, while my K-180WLA just uses an ordinary wire loop connected directly to the amplifier with no isolation, just like the MLA-30. Schematic [here](https://www.g8jnj.net/activeantennas.htm#MLA30).

I have a Nooelec Flamingo FM bandstop filter which I will connect to the YouLoop, then a LaNA HF LNA (which I've also ordered) below the filter for better linearity. Hopefully this combination will perform better on HFDL for me than my present active loop. I'm also getting 3 meters of solud AWG10 copper wire with which to replace the stainless steel 55cm diameter loop of my current K-180WLA antenna. The replacement loop will be around 95cm and use larger-diameter wire which may improve signal gain.
